src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareService.java

```diff
@@ -78,54 +84,99 @@ public EvidenceShareService(
-                return Optional.empty();
+                log.info("Skipping bulk print. Sending straight to {} state for case id {}",
+                    EventType.SENT_TO_DWP.getCcdType(), sscsCaseDataCallback.getCaseDetails().getId());
+                ccdService.updateCase(caseData, Long.valueOf(caseData.getCcdCaseId()),
+                    EventType.SENT_TO_DWP.getCcdType(), SENT_TO_DWP, bulkPrintInfo.getDesc(),
+                    idamService.getIdamTokens());
             }
+            return true;
         }
-        log.info("Feature flag turned off for sending to DWP. Skipping evidence share for case id {}", sscsCaseDataCallback.getCaseDetails().getId());
-        if (sscsCaseDataCallback.getCaseDetails().getState().toString().equals(State.VALID_APPEAL.toString())) {
-            log.info("Sending case back to appeal created so it is in correct state for old workflow for case id {}", sscsCaseDataCallback.getCaseDetails().getId());
-            ccdService.updateCase(caseData, Long.valueOf(caseData.getCcdCaseId()), EventType.MOVE_TO_APPEAL_CREATED.getCcdType(), "Case created", "Sending back to appealCreated state", idamService.getIdamTokens());
-        }
+        return false;
+    }
 
-        return Optional.empty();
+    private BulkPrintInfo bulkPrintCase(Callback<SscsCaseData> sscsCaseDataCallback) {
+        SscsCaseData caseData = sscsCaseDataCallback.getCaseDetails().getCaseData();
+        if (isAllowedReceivedTypeForBulkPrint(sscsCaseDataCallback.getCaseDetails().getCaseData())) {
+            log.info("Processing callback event {} for case id {}", sscsCaseDataCallback.getEvent(),
+                sscsCaseDataCallback.getCaseDetails().getId());
+
+            DocumentHolder holder = documentRequestFactory.create(
+                sscsCaseDataCallback.getCaseDetails().getCaseData(),
+                sscsCaseDataCallback.getCaseDetails().getCreatedDate());
+
+            if (holder.getTemplate() != null) {
+                log.info("Generating DL document for case id {}", sscsCaseDataCallback.getCaseDetails().getId());
+
+                IdamTokens idamTokens = idamService.getIdamTokens();
+                documentManagementServiceWrapper.generateDocumentAndAddToCcd(holder, caseData, idamTokens);
+                List<Pdf> existingCasePdfs = toPdf(caseData.getSscsDocument());
+
+                log.info("Sending to bulk print for case id {}", sscsCaseDataCallback.getCaseDetails().getId());
+                caseData.setDateSentToDwp(LocalDate.now().toString());
+                return BulkPrintInfo.builder()
+                    .uuid(bulkPrintService.sendToBulkPrint(existingCasePdfs, caseData).orElse(null))
+                    .allowedTypeForBulkPrint(true)
+                    .desc(buildEventDescription(existingCasePdfs))
+                    .build();
+            }
+            return null;
```

Comment:
When this is null, should we not be flagging the error as a secondary state on the case? At the moment you are only flagging it if there is an exception thrown. I guess there could be a problem getting the template so probably need to capture this scenario as well.

Reply:
ok. I was respecting the logic as it was previously, but yeah, it makes sense to flag the error for this case too.

Reply:
done